### PR TITLE
imp_url: normalize path for default out name

### LINKED ISSUE #2237: import: strange error with path ending in `/`
```
DVC version: 0.50.1
Python version: 3.7.3
Platform: Darwin-18.6.0-x86_64-i386-64bit
```

> First `dvc init` an empty DVC repo and run the following inside. In my case the working dir is called `emptydvcrepo`.

```console
$ dvc import https://github.com/iterative/example-get-started data/prepared/
...
ERROR: failed to import 'data/prepared/' from
'https://github.com/iterative/example-get-started'. - current working directory
'/path/to/emptydvcrepo' is specified as an output in 'emptydvcrepo.dvc'. Use another CWD
to prevent any data removal.
```

Notice the given `path`, `data/prepared/` ending with `/` and the error mentioning an inexistent DVC-file called `emptydvcrepo.dvc`.

It works normally without the slash.

### PR DESCRIPTION
**Pull request by pared**
* [x] Have you followed the guidelines in our
      [Contributing document](https://dvc.org/doc/user-guide/contributing)?

* [x] Does your PR affect documented changes or does it add new functionality
      that should be documented? If yes, have you created a PR for
      [dvc.org](https://github.com/iterative/dvc.org) documenting it or at
      least opened an issue for it? If so, please add a link to it.

-----
Fixes #2237

### CODE REVIEW

File: dvc/repo/imp_url.py [hunk 1]
```diff
@@ -10,7 +10,7 @@ def imp_url(
 ):
     from dvc.stage import Stage
 
-    default_out = os.path.basename(urlparse(url).path)
+    default_out = os.path.basename(os.path.normpath(urlparse(url).path))
```
Comment (Suor) [inline dvc/repo/imp_url.py]:
```python
default_out = URLInfo(url).name
```

**Comment by efiop** [comment]
@pared Just a heads up: tests failed.

File: dvc/repo/imp_url.py [hunk 2]
```diff
@@ -10,7 +8,7 @@ def imp_url(
 ):
     from dvc.stage import Stage
 
-    default_out = os.path.basename(urlparse(url).path)
+    default_out = URLInfo(url).name
     out = out or default_out
```
Comment (Suor) [inline dvc/repo/imp_url.py]: `out = out or URLInfo(url).name`
No point in parsing it if we have out specified.

File: tests/func/test_import_url.py [hunk 3]
```diff
@@ -46,7 +46,7 @@ def test(self):
 
 
 class TestFailedImportMessage(TestDvc):
-    @patch("dvc.repo.imp_url.urlparse")
+    @patch("dvc.repo.imp_url.URLInfo")
```
Comment (Suor) [inline tests/func/test_import_url.py]:
```suggestion
    @patch("dvc.repo.imp_url.URLInfo", side_effect=Exception("message"))
```
And drop side effect handling in func body.

**Comment by Suor** [comment]
Also, about test fails. Using `URLInfo` didn't end that well) I added a protection there. I really wanted to test that outs and remotes always have resolved urls. As far as I see now there is now simple way to move it out, so I propose reverting to `urlparse()` for now, use `PurePosixPath(path).name` instead of a fs util though.

**Comment by efiop** [comment]
tests still fail :) @pared don't you run tests locally before submitting?

**Comment by pared** [comment]
@efiop sorry, mentally it was "quick fix" and it "surely will pass now" Ill take a closer look now.

**Comment by efiop** [comment]
@pared Tests failed on windows

**Comment by efiop** [comment]
@Suor @pared let's just go with the original simple approach, no need to mess around with PathInfos

**Comment by pared** [comment]
@efiop weren't we switching to using pathlib objects?

**Comment by efiop** [comment]
@pared in that specific part I don't think it is needed, since, as we've seen, it introduces a lot of confusion.

**Pull request merged**